fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -67,11 +68,62 @@ internal class HttpClient(private val proxy: Proxy? = null) : Client {
         }
     }
 
+    override suspend fun awaitRequest(request: Request): Response = suspendCoroutine { continuation ->
+        try {
+            val connection = establishConnection(request) as HttpURLConnection
+            connection.apply {
+                connectTimeout = Fuel.testConfiguration.coerceTimeout(request.timeoutInMillisecond)
+                readTimeout = Fuel.testConfiguration.coerceTimeoutRead(request.timeoutReadInMillisecond)
+                doInput = true
+                useCaches = false
+                requestMethod = if (request.method == Method.PATCH) Method.POST.value else request.method.value
+                instanceFollowRedirects = false
+
+                for ((key, value) in request.headers) {
+                    setRequestProperty(key, value)
+                }
+
+                if (request.method == Method.PATCH) {
+                    setRequestProperty("X-HTTP-Method-Override", "PATCH")
+                }
+
+                setDoOutput(connection, request.method)
+                setBodyIfDoOutput(connection, request)
+            }
+
+            val contentEncoding = connection.contentEncoding ?: ""
+
+            continuation.resume(Response(
+                    url = request.url,
+                    headers = connection.headerFields.filterKeys { it != null },
+                    contentLength = connection.contentLength.toLong(),
+                    statusCode = connection.responseCode,
+                    responseMessage = connection.responseMessage.orEmpty(),
+                    dataStream = try {
+                        val stream = connection.errorStream ?: connection.inputStream
+                        if (contentEncoding.compareTo("gzip", true) == 0) GZIPInputStream(stream) else stream
+                    } catch (exception: IOException) {
+                        try {
+                            connection.errorStream ?: connection.inputStream?.close()
+                        } catch (exception: IOException) {
+                        }
+                        ByteArrayInputStream(ByteArray(0))
+                    }
+            ))
+        } catch (exception: Exception) {
```

Comment:
Do we have tests to all possible exceptions that can be thrown here?

Reply:
By looking https://codecov.io/gh/kittinunf/Fuel/src/master/fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt, not much.

Reply:
@lucasvalenteds sorry if I have missed a step here but I thought that all exceptions would be caught in the block as they all inherit from Exception?

I am sorry if I have misunderstood the problem

Reply:
@markGilchrist The question is if there are tests that verify the subtypes. For exemple, if a function inside the `try {}` block can throw a `IllegalArgumentException`, is there a test that demonstrate that happening?